refactor: hordelib now single threaded, no more need for `.patch`; feat: SDXL support

tazlin opened the pull request:
- **feat**: Remove need for the `.patch` file used to modify ComfyUI
  - The behavior introduced with the `.patch` file was largely focused on thread safety. A new multi-processing model is replacing this approach, eliminating the need for the deeply time consuming process and confusing process of authoring a compatibility patch each ComfyUI release.
- **feat**: SDXL support
- **fix**: `VAEDecodeTiled` no longer used by default
  - I have been able to demonstrate that the tiled decoder has substantial performance issues in certain circumstances, and currently ComfyUI is handling the decision as to when to use it.
- **feat**: multiprocessing support
- **refactor**: Model managers no longer directly load models in and out of ram (this is handled by comfy internally)
- **refactor**: Model managers considerably cleaned up, many unused or dead-end functions remove, and numerous readability and usability fixes made.
- **style**: enforced more lint rules (`UP`, `RET`, `B`)
- **CI**: implemented pre-commit (use locally with `pre-commit install`, any attempt to commit will run black/ruff/mypy/etc).
- **build**: images are no longer packaged in the `.tar.gz` or `.whl` distributions 
  - A warning has been added to the tests if the folder cannot be found.

Closes https://github.com/Haidra-Org/hordelib/issues/32, Closes https://github.com/Haidra-Org/hordelib/issues/31, Closes #49, Closes #27

tazlin committed:
tests: feat: show img pair which mismatched

tazlin committed:
style: black formatter catchup

tazlin committed:
fix: Remove `VAEDecodeTiled`; switch to ` VAEDecode`

tazlin committed:
fix: remove references to now unused clip/blip code

tazlin committed:
fix: don't sha validate annotators if previously done

tazlin committed:
feat: give better in-progress inference log information

tazlin committed:
fix: downgrade warning for (re) loading model managers

tazlin committed:
fix: remove any load to ram/vram code, make hordelib single threaded

tazlin committed:
tests: confirm deliberate+sdxl 1.0 are downloaded and valid on test

tazlin committed:
ci: setup env correctly for CI

tazlin committed:
fix: remove inpainting sdxl tests; update sdxl img2img test

tazlin committed:
feat: display time too in `run_kudos_test`

tazlin committed:
fix: black version in tox

tazlin committed:
tests: confirm lora+hires work together

tazlin committed:
refactor: fix variable typos/update to snake case

tazlin committed:
tests: download all required models/files for test upfront

tazlin committed:
build: better build directives

tazlin committed:
tests: account for more lora circumstancces

tazlin committed:
tests: include one of each SDXL suggested resolution

tazlin committed:
fix: malformed regex for comfyui inference progress detection

tazlin committed:
fix: add title to facerestore node

tazlin committed:
refactor: better `BaseModelManager` func names, remove unused funcs

tazlin committed:
style: reintroduce imported but unused lint rule

tazlin committed:
style: adds precommit; applies all whitespace fixes

tazlin committed:
chore: ignore big style fix from git blame

tazlin committed:
ci: trigger pre-commit on CI

tazlin committed:
refactor: more paths to `Path` objects; more doc strings in base.py

tazlin committed:
refactor: comfy handles memory; better multiprocessing scaffolding

Also brings a number of systems in line with other preceding refactors

tazlin committed:
fix: remove references from `hordelib.__init__` (breaks build)

tazlin committed:
tests: fix: swapped dimensions for sdxl test

tazlin committed:
tests: fix: make lora last_use check less likely to fail

tazlin committed:
fix: bring `image_upscale` + `image_facefix` inl ine with return type rework

tazlin committed:
fix: cleaner progress bar comfy intercept replay/display

tazlin committed:
feat: process tracking support in logging

tazlin committed:
tests: default to showing all debug messages during CI

tazlin committed:
fix: hide debug message

tazlin committed:
style: ruff/black catchup

tazlin committed:
style: missed files on last style catchup

tazlin committed:
ci: add missing dep for mypy pre-commit hook

tazlin committed:
chore: ignore big style catchup in git blame

tazlin committed:
build: do not include images folder in pypi packages

tazlin committed:
tests: fail tests if `images_expected` folder is missing

tazlin committed:
fix: missing dep in reqs.txt

tazlin committed:
style: enforce lint rules `UP` and `RET`

tazlin committed:
style: enforce lint ruleset `B` (bugbear)

tazlin committed:
tests: issue with test order; more leniency to lora last use check

tazlin committed:
fix: errant reference to removed download callback

tazlin committed:
fix: better handle missing downloads/sha256 files

tazlin committed:
fix: correct version pins

tazlin committed:
fix: loras not updating last use

tazlin committed:
fix: quiet some Comfy_Horde logging

db0 commented:
Epic. Question, you removed the blip/clip code. How will the safety check be handled in the future?

tazlin commented:
It's all going to happen in horde_safety.

tazlin committed:
fix: further attempts to minimize threading issues with loras

tazlin committed:
fix: remove unused model reference download code from base.py

tazlin committed:
fix: better log messages for multiproc, quiet comfy init `print(...)`s

tazlin committed:
style: fix line too long

tazlin committed:
feat: better logging, fix TI missing from payload

tazlin committed:
fix: lora test case should be lowercase

tazlin committed:
fix: none handling in lora `is_model_available`

tazlin committed:
Merge branch 'main' into regen

tazlin committed:
feat: latest comfyui

tazlin committed:
fix: remove old references to comfy vars

db0 commented:
It's happening!

tazlin committed:
fix: validate TIs correctly (as with loras)

tazlin committed:
fix: don't crash on invalid model reference json

tazlin committed:
fix: default to loading all model managers (is now more lightweight)

The overhead of loading all the model managers is much less now, and a more developer friendly stance is to load all of them by default.

tazlin committed:
feat: better log messages during inference

tazlin committed:
fix: unload models more reliably when switching

tazlin committed:
style: lint fix

tazlin committed:
tests: give TI model manager tests 3 chances at a download

tazlin committed:
feat: tiling

tazlin committed:
fix: remove indev memory usage hacks

tazlin committed:
feat: better tiling support

- see `add_circular_option_to_conv_2d` in `node_model_loader.py` for more info
- also includes tests for varies modes along side tiling

tazlin committed:
refactor: more idiomatic seamless tiling

tazlin committed:
feat: worker beta changes

tazlin committed:
chore: version bump, min requirements

tazlin committed:
fix: accurate download script

tazlin committed:
fix: handle base64 for source image

tazlin committed:
fix: mask image

tazlin committed:
fix: handle pp when given sdk payload: adds rembg

tazlin committed:
fix: remove stats log level

tazlin committed:
fix: cleanup logger sink logic

tazlin committed:
fix: lora download hang-up; don't hard crash on lora dl fail

tazlin committed:
fix: strip() orig name for TI on model reference load

tazlin committed:
tests: take 2 on TI test

tazlin committed:
fix: be more consistent with TI model ref keys

tazlin committed:
fix: bad `strip_background()` func signature

tazlin committed:
fix: dup func def in logger.py

tazlin committed:
tests: fail on first controlnet mismatch

tazlin committed:
tests: now detect runtime error on unknown pipeline

tazlin committed:
feat: comfyui 099226015edcdf595d915f0f39359ef0d83fbba6

tazlin committed:
feat: comfyui callback; better type hints

See hordelib.py ` -775,7 +787,7 @@ class HordeLib:` for a presumed bug fix

tazlin committed:
fix: even more defensive programing with TIs

tazlin committed:
tests: will somebody rid me of this turbulent test

tazlin committed:
fix: don't crash on `comfyui_callback` kwarg for HordeLib()

tazlin committed:
fix: don't crash on model unload
